File: week1/day3/algo-linear-search/js/linearSearch.js
const arrayToSearchThrough = [];
for (let i = 1; i <= 1000; i++) {
    arrayToSearchThrough.push(i);
}

exports.linearSearch = function(valueToFind, arrayToSearchThrough) {

    for (let i = 0; i < arrayToSearchThrough.length; i++) {
        if (valueToFind === arrayToSearchThrough[i]){
            return i
        }
    }

    return undefined
};

exports.linearSearchGlobal = function(valueToFind, arrayToSearchThrough) {
    
    result = []

    for (let i = 0; i < arrayToSearchThrough.length; i++) {
        if (valueToFind === arrayToSearchThrough[i]) {
            result.push(i)
        }
    }

    if (result.length > 0) {
        return result
    }

    return undefined
};

File: week1/day3/algo-linear-search/python/linear_search.py
array_to_search_through = []
for number in range(1, 1001):
    array_to_search_through.append(number)

def linear_search(value_to_find, array_to_search_through):
    count = 0

    for value in array_to_search_through:

        if value == value_to_find:
            return count

        count += 1

def linear_search_global(value_to_find, array_to_search_through):
    count = 0
    result = []

    for value in array_to_search_through:

        if value == value_to_find:
            result.append(count)

        count += 1

    return result